### ISSUE TITLE
Add wire stubs to black box vertices

### ISSUE THREAD

**Issue by arouinfar** [phetsims/circuit-construction-kit-black-box-study#21]
After initial student interviews, Shima reported:
> In particular for low content knowledge participants, it would be really helpful and somehow necessarily to bring back the wires sticking out of the black box. In the recent version, participants only see the terminals as four dashed circle, in the previous version there were four wires coming out of the black box. The current version without wires for terminals, made it spatially and conceptually hard for participants to understand the goal of the activity. 4 out of 5 low content knowledge interviewees perceived the activity as they have to build something on top of the blackbox. Not having the wire coming out of black box is definitely not the only reason for this misunderstanding as I mentioned in the last email, but having the wire could be helpful in discouraging this misunderstanding.

Shima's timeline:
>  I am using CCK-BB for pre-test of a study at Stanford, starting January 9th.That would be great to address #2 before that round. would that be possible? Besides that, I will use CCK-BB for a study in high school February and beginning of March. For this study, I really need #2 to be fixed, as with high schoolers I want the activity to be the least confusing. Furthermore, I will use CCK-BB again for the post-test of Stanford study. 

I'm imagining the wire stubs would look like this, and if possible, the node between the black box and wire would not be severable. 
<img width="512" alt="screen shot 2016-12-20 at 1 03 17 pm" src="https://cloud.githubusercontent.com/assets/8419308/21364037/b7cce33e-c6b4-11e6-8fd8-8a35a53883f0.png">

@samreid I'm not sure what you holiday travel and priorities look like for the next few weeks, but do you think this is something that can be addressed before Jan 9th?

**Comment by samreid** [comment]
My rough guess (before working on it) is that it will likely take 2 days (16 developer hours) to prepare this for the first round of testing.  @kathy-phet, @ariel-phet, @arouinfar should I go for it?  Will @phet-steele or other QA team members be available for testing this in advance of Jan 9?

**Comment by arouinfar** [comment]
@samreid, I don't believe this will need full QA testing, but I think it would be good for @phet-steeleand I to go have a go at it.

**Comment by kathy-phet** [comment]
@samreid. I would say go for it. I think it is a feature that does help, and we would want to use in any CCK: BB design we release ourselves. I would say they need to be fixed and unpickable, so that they cannot be stretched or moved or deleted.

**Comment by samreid** [comment]
Sounds good, reassigned to me and I'll keep you apprised of my progress.

**Comment by samreid** [comment]
In the above commit, I added support for the "simplest thing that might possibly work".  You can test it on phettest.colorado.edu using VPN.  It looks like this:

![image](https://cloud.githubusercontent.com/assets/679486/21464470/481d630e-c93b-11e6-9f92-f2acd8354741.png)

I noted the following:
* the black box wires are gray
* the vertex at the black box is not shown (perhaps this is ok)
* you cannot see the electrons in the black box wires
* When pressing test, the vertices are shown like this:

![image](https://cloud.githubusercontent.com/assets/679486/21464479/88850c58-c93b-11e6-8fdd-e28e27e29627.png)

You can still attach your own leader wires like so:

![image](https://cloud.githubusercontent.com/assets/679486/21464480/9b472736-c93b-11e6-85bd-c8915c06421c.png)

But you can also put pieces like this:

![image](https://cloud.githubusercontent.com/assets/679486/21464484/ad9194ee-c93b-11e6-9319-743153a36d0a.png)

Keeping the circuit as similar as possible between the black box internals and externals has been important in simplifying the implementation.  I thought this would be a good point to get feedback from @arouinfar to see what steps I should take next, if any.

**Comment by arouinfar** [comment]
@samreid can you publish this as a dev version? I'd like to reach out directly to Shima to see if this would be a reasonable solution for the purposes of her study.

**Comment by arouinfar** [comment]
Pinging @samreid on this issue. I'd like to let Shima take a look at this in a dev version to see if this would be an acceptable solution.

**Comment by samreid** [comment]
Thanks for the reminder @arouinfar, I'll work on this shortly.

**Comment by samreid** [comment]
I pushed "deploy" for 1.0.5-phetio but https://phet-io.colorado.edu/sims/circuit-construction-kit-black-box-study/1.0.5-phetio is not appearing.

**Comment by samreid** [comment]
Following the instructions on https://github.com/phetsims/phet-info/blob/master/deployment-info/doc/deploy-rc-phet-io.md I used `sudo journalctl -fu build-server`.  It told me:

```
-- Logs begin at Wed 2016-12-21 23:00:59 MST. --
Dec 29 15:37:56 phet-server.int.colorado.edu build-server[144681]: },
Dec 29 15:37:56 phet-server.int.colorado.edu build-server[144681]: "sun": {
Dec 29 15:37:56 phet-server.int.colorado.edu build-server[144681]: "sha": "2f55106098902caa6c271e79df5a28380e8c78a4",
Dec 29 15:37:56 phet-server.int.colorado.edu build-server[144681]: "branch": "HEAD"
Dec 29 15:37:56 phet-server.int.colorado.edu build-server[144681]: },
Dec 29 15:37:56 phet-server.int.colorado.edu build-server[144681]: "tandem": {
Dec 29 15:37:56 phet-server.int.colorado.edu build-server[144681]: "sha": "005c5582da1cba304d5d650533e6059eae197c6b",
Dec 29 15:37:56 phet-server.int.colorado.edu build-server[144681]: "branch": "HEAD"
Dec 29 15:37:56 phet-server.int.colorado.edu build-server[144681]: }
Dec 29 15:37:56 phet-server.int.colorado.edu build-server[144681]: }
```

**Comment by samreid** [comment]
Higher up (with --lines=200) it says:

> Error: Command failed: /bin/sh -c grunt checkout-shas --buildServer=true --repo=circuit-construction-kit-black-box-study

**Comment by samreid** [comment]
I tried running the command locally and it failed like so:

```
~/github/circuit-construction-kit-black-box-study$ grunt checkout-shas --buildServer=true --repo=circuit-construction-kit-black-box-study
Running "checkout-shas" task
Warning: Unable to read "../chipper/js/build-server/tmp/dependencies.json" file (Error code: ENOENT). Use --force to continue.
```

I tried without the --buildServer=true, and it worked OK.

**Comment by samreid** [comment]
I asked for help on the Skype channel, and will try running the deploy command again.

**Comment by samreid** [comment]
The new version was deployed as 1.0.5-phetio.  You can use all of the same links we sent Shima originally, but change the 1.0.4 to 1.0.5.  For instance:

https://phet-io.colorado.edu/sims/circuit-construction-kit-black-box-study/1.0.5-phetio/circuit-construction-kit-black-box-study_en-phetio.html?phet-io.standalone&phet-io.log=lines

@arouinfar please double-check this before passing it along.

**Comment by arouinfar** [comment]
Thanks @samreid! I've sent an email to Shima.

**Comment by arouinfar** [comment]
Shima liked the solution proposed by @samreid in https://github.com/phetsims/circuit-construction-kit-black-box-study/issues/21#issuecomment-269062104. She responded:

> Thanks Amy and Sam
I like the solution, it looks great. Playing with it, I think it is more intuitive now. Hopefully, that would be the case for the less knowledgable participants as well.
I will keep you posted about the interviews and how they receive the solution.

@samreid we should do some testing before Shima uses the sim with students. Is 1.0.5-phetio an acceptable version to test?

**Comment by samreid** [comment]
@arouinfar this version was built as a minimal delta, and only differs with respect to the wires (no changes in common code libraries were required).  This was done so that only minimal testing would be necessary.  Still, you or @phet-steele should take the changes for a test drive.

**Comment by arouinfar** [comment]
Thanks @samreid. I will create a task for testing and will coordinate with @phet-steele. However, I don't believe the default values/tweaker increments discussed in https://github.com/phetsims/circuit-construction-kit-common/issues/221 have been migrated over to 1.0.5-phetio. I can create separate issue for that, but I also don't know if it merits releasing a new version for testing, or not.

**Comment by samreid** [comment]
> However, I don't believe the default values/tweaker increments discussed in phetsims/circuit-construction-kit-common#221 have been migrated over to 1.0.5-phetio. 

That is correct.  However, if it is not something critical for Shima's study, then I'd be inclined to leave it out.  Or, if we end up delivering another version beyond 1.0.5-phetio, then we could make that change then.

**Comment by kathy-phet** [comment]
I think it is a usability issue not to be able to return to the default values even if it did not come up in Shima's original interviews - so I think we need to make that change in the version we deliver to Shima. 

I just heard from Argenta a few minutes ago that Shima and her will be doing a classroom trial with CCK - Black Box and the first screen of CCK in about a month. I think in that context it will be important to have this improvement of the value behaviors there.

**Comment by samreid** [comment]
> I think it is a usability issue not to be able to return to the default values even if it did not come up in Shima's original interviews - so I think we need to make that change in the version we deliver to Shima.

Sure, I'll work on that.  @arouinfar and @phet-steele can hold off testing until 1.0.6-phetio is ready.

**Comment by samreid** [comment]
Added 0.5 patch into https://phet-io.colorado.edu/sims/circuit-construction-kit-black-box-study/1.0.6-phetio/circuit-construction-kit-black-box-study_en-phetio.html?phet-io.standalone&phet-io.log=lines

@arouinfar can you test or schedule @phet-steele for testing?

**Comment by arouinfar** [comment]
Thanks @samreid! I'll make a testing issue in the tasks repo.

**Comment by arouinfar** [comment]
@samreid while verifying https://github.com/phetsims/circuit-construction-kit-black-box-study/issues/21#issuecomment-269062104 for https://github.com/phetsims/tasks/issues/761, I discovered that solder joints appear on the wire stubs after a component is removed from the toolbox. 

On startup:
<img width="438" alt="screen shot 2017-01-05 at 11 52 48 am" src="https://cloud.githubusercontent.com/assets/8419308/21693641/eb11e4ba-d33e-11e6-92f5-9292797e1041.png">

After dragging any component out of the toolbox:
<img width="462" alt="screen shot 2017-01-05 at 11 52 57 am" src="https://cloud.githubusercontent.com/assets/8419308/21693640/eb0aa1dc-d33e-11e6-80e5-28869aab8632.png">

**Comment by arouinfar** [comment]
@samreid I haven't noticed any other issues with the wire stubs thus far, and I don't think this is a particularly blocking issue. If nothing else crops up from testing, I think Shima can use 1.0.6 for the pre-test data she's collecting on Monday.

**Comment by samreid** [comment]
Excellent, I'll leave it up to you to coordinate with Shima regarding delivering the new version.

**Comment by arouinfar** [comment]
@samreid I double checked with Shima that the wire stubs are acceptable for the high school study as well. I'll let you know when I hear back.

Assuming there aren't any other change requests, the only thing that's left to address for this issue would be the solder joints I described in https://github.com/phetsims/circuit-construction-kit-black-box-study/issues/21#issuecomment-270730658.

**Comment by arouinfar** [comment]
Leaving self-assigned until we hear back from Shima.

**Comment by arouinfar** [comment]
@samreid after re-reviewing the wire stub behavior Shima responded:

>Amy I believe none of the above issues are necessarily critical except for:
*Electrons are not visible in the black box wires
This can cause misconception. So, if there is one of these compromises that  should be resolved for sure is the above one. 
About the other issues:
I am not sure if I understand this point exactly:
*The vertex at the black box is not shown
Also, I do not like the floating terminals in the test tab very much as well, why cannot we have four plain wire attaching to them and going inside black box? If it is there is technical reason for it, which I have probably limited knowledge about, then I can make the four floating terminals work for students by giving some more explanations to them when they want to use the test tab.
Thanks again and sorry for delay,
Shima

I responded to Shima to clarify that the "vertex at the black box is not shown" refers to https://github.com/phetsims/circuit-construction-kit-black-box-study/issues/21#issuecomment-270730658. I've also asked for an updated timeline. Last I heard, Shima did not have a precise starting date for the classroom study, but I've replied to ask her for a general estimate of when in February (early vs. late) they'll need a new version.


To summarize Shima's change requests/priorities
* Show the electrons in the wire stubs (critical)
* Add wires to floating terminals in the Test scene (important, but can be worked around)

**Comment by samreid** [comment]
Based on the preceding comment, I'll create 2 new issues and close this one.

**Comment by samreid** [comment]
New issues are above, closing.
